presto-main/src/main/java/com/facebook/presto/sql/planner/SystemPartitioningHandle.java

```diff
@@ -44,24 +44,108 @@
 public final class SystemPartitioningHandle
         implements ConnectorPartitioningHandle
 {
+    public static Partitioning singlePartition()
+    {
+        return createSystemPartitioning(SystemPartitioning.FIXED, SystemPartitionFunction.SINGLE);
```

Comment:
The definition of SystemPartitionFunction.SINGLE implies partitioning on the empty of columns. We need to make sure that property is retained.